pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -304,10 +304,12 @@ cdef class _Timestamp(datetime):
         out = get_date_field(np.array([val], dtype=np.int64), field)
         return int(out[0])
 
-    cpdef _get_start_end_field(self, field):
+    cpdef bint _get_start_end_field(self, str field):
```

Comment:
so why is this in _Timestamp again (as opposed to Timestamp); this is why its cpdef, why not just cdef?

Reply:
Not sure why its in _Timestamp instead of Timestamp (though my understanding is putting it in _Timestamp is slightly more performant and smaller memory footprint); it was that way before I got here.

It is `cpdef` and not `cdef` because if it were cdef then calling it from Timestamp would be an AttributeError.  That's why #18446 moved a bunch of properties up to _Timestamp after making it cdef.